soc/xtensa/intel_adsp/common/rimage_modules.c

```diff
@@ -12,12 +12,19 @@
  * left unchanged for historical purposes.
  */
 
+/* For BRNGUP or BASEFW module, two UUIDs are allowed for now, one for
+ * APL, and one for cAVS 1.8+ platforms, and for IPC4, the module UUID
+ * in manifest must be identical with the one in component. We use UUIDs
+ * for cAVS 1.8+ platforms here because there will be *NO* IPC4 support
+ * for APL, and thus, there will be no BRNGUP and BASEFW modules in APL
+ * firmware.
```

Comment:
These modules are already in APL firmware right now:

```
sof/tools/sof_ri_info/sof_ri_info.py  zephyrproject/build-apl/zephyr/zephyr.ri

  cavs0015 (ADSP Manifest) file offset 0x22e0 name ADSPFW build ver 0.0.0.0 feature mask 0xffff image flags 0x0
    HW buffers base address 0x0 length 0x0
    Load offset 0x2000

    BRNGUP    0d7b48cc-1ea9-470a-a8c1-533424528a17
      entry point 0xb000a000 type 0x21 ( loadable LL )
      cfg offset 0 count 0 affinity 0x3 instance max count 1 stack size 0x1
      .text   0xb000a000 file offset 0x8000 flags 0x1001f ( contents alloc load readonly code type=0 pages=1 )
      .rodata 0xffffffff file offset 0x9000 flags 0x1012f ( contents alloc load readonly data type=1 pages=1 )
      .bss    0x0 file offset 0x0 flags 0xf00 ( type=15 pages=0 )

    BASEFW    fc869e2e-45f8-4045-a416-89880ae320a9
      entry point 0xbe008400 type 0x21 ( loadable LL )
      cfg offset 0 count 0 affinity 0x3 instance max count 1 stack size 0x1
      .text   0xbe008000 file offset 0x9000 flags 0x2d001f ( contents alloc load readonly code type=0 pages=45 )
      .rodata 0xbe035000 file offset 0x36000 flags 0x1d012f ( contents alloc load readonly data type=1 pages=29 )
      .bss    0xbe052000 file offset 0x0 flags 0x20202 ( alloc type=2 pages=2 )



```

Reply:
Also, can you update the existing comment instead of adding a new one?

Reply:
They are there because the information is read from manifest. the module information is inserted to manifest by rimage, but actually, there are no module code in firmware for SOF with IPC3. But for SOF with IPC4, the modules both exist in manifest and firmware code.